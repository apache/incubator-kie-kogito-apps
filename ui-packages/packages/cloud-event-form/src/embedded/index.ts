export * from './EmbeddedCloudEventForm';
export * from './EmbeddedCloudEventFormChannelApiImpl';
