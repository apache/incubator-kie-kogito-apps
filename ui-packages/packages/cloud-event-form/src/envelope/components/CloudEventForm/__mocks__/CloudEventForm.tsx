import React from 'react';

const MockedCloudEventForm = (): React.ReactElement => {
  return <></>;
};

export default MockedCloudEventForm;
