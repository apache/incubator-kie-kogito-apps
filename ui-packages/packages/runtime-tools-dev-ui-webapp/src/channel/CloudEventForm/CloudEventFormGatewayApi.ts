import { CloudEventRequest } from '@kogito-apps/cloud-event-form';
import { triggerCloudEvent, triggerStartCloudEvent } from '../apis';

export interface CloudEventFormGatewayApi {
  triggerStartCloudEvent(event: CloudEventRequest): Promise<string>;
  triggerCloudEvent(event: CloudEventRequest): Promise<any>;
}

export class CloudEventFormGatewayApiImpl implements CloudEventFormGatewayApi {
  constructor(private readonly baseUrl: string) {}

  async triggerStartCloudEvent(event: CloudEventRequest): Promise<string> {
    const response = await triggerStartCloudEvent(event, this.baseUrl);
    return response;
  }

  triggerCloudEvent(event: CloudEventRequest): Promise<any> {
    return triggerCloudEvent(event, this.baseUrl);
  }
}
