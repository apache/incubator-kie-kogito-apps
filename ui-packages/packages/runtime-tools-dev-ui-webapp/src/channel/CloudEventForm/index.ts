export { CloudEventFormGatewayApi } from './CloudEventFormGatewayApi';
export * from './CloudEventFormContext';
export { default as CloudEventFormContextProvider } from './CloudEventFormContextProvider';
