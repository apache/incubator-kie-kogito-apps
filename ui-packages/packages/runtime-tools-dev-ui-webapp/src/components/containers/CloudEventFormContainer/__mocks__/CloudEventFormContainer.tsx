import React from 'react';

const MockedCloudEventFormContainer = ({
  onSuccess,
  onError
}): React.ReactElement => {
  React.useEffect(() => {
    onSuccess();
    onError();
  }, []);
  return <></>;
};

export default MockedCloudEventFormContainer;
