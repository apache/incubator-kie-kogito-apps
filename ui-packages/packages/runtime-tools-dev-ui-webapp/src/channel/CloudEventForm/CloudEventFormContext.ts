import React, { useContext } from 'react';
import { CloudEventFormGatewayApi } from './CloudEventFormGatewayApi';

const CloudEventFormContext =
  React.createContext<CloudEventFormGatewayApi>(null);

export const useCloudEventFormGatewayApi = (): CloudEventFormGatewayApi =>
  useContext<CloudEventFormGatewayApi>(CloudEventFormContext);

export default CloudEventFormContext;
