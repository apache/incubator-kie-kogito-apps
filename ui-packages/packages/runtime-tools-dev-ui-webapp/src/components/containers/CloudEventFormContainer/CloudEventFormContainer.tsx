import React, { useCallback } from 'react';
import {
  componentOuiaProps,
  OUIAProps
} from '@kogito-apps/ouia-tools/dist/utils/OuiaUtils';
import { useCloudEventFormGatewayApi } from '../../../channel/CloudEventForm';
import { EmbeddedCloudEventForm } from '@kogito-apps/cloud-event-form/dist/embedded';
import { CloudEventRequest } from '@kogito-apps/cloud-event-form/dist/api';
import { useParams } from 'react-router';
import { useDevUIAppContext } from '../../contexts/DevUIAppContext';

export type CloudEventFormContainerProps = {
  isTriggerNewInstance: boolean;
  onSuccess: (id: string) => void;
  onError: (details?: string) => void;
};

export type CloudEventFormContainerParams = {
  instanceId?: string;
};

const CloudEventFormContainer: React.FC<
  CloudEventFormContainerProps & OUIAProps
> = ({ isTriggerNewInstance, onSuccess, onError, ouiaId, ouiaSafe }) => {
  const gatewayApi = useCloudEventFormGatewayApi();
  const appContext = useDevUIAppContext();

  const { instanceId } = useParams<CloudEventFormContainerParams>();

  const triggerStartCloudEvent = useCallback(
    (event: CloudEventRequest) => {
      return gatewayApi
        .triggerStartCloudEvent(event)
        .then((businessKey) => {
          onSuccess(
            `A workflow with business key ${businessKey} has been successfully triggered.`
          );
        })
        .catch((error) => handleError(error));
    },
    [gatewayApi, onSuccess, onError]
  );

  const triggerCloudEvent = useCallback(
    (event: CloudEventRequest) => {
      return gatewayApi
        .triggerCloudEvent(event)
        .then((response) => {
          console.log(response);
          onSuccess('The CloudEvent has been successfully triggered.');
        })
        .catch((error) => handleError(error));
    },
    [gatewayApi, onSuccess, onError]
  );

  const handleError = useCallback(
    (error) => {
      const message =
        error?.message ||
        'Unknown error. More details in the developer tools console.';
      onError(message);
    },
    [gatewayApi, onSuccess, onError]
  );

  return (
    <EmbeddedCloudEventForm
      {...componentOuiaProps(ouiaId, 'cloud-event-form-container', ouiaSafe)}
      targetOrigin={appContext.getDevUIUrl()}
      isNewInstanceEvent={isTriggerNewInstance}
      defaultValues={{
        cloudEventSource: '/local/quarkus-devUi',
        instanceId: instanceId ?? undefined
      }}
      driver={{
        triggerCloudEvent(event: CloudEventRequest): Promise<void> {
          const doTrigger = isTriggerNewInstance
            ? triggerStartCloudEvent
            : triggerCloudEvent;
          return doTrigger(event);
        }
      }}
    />
  );
};

export default CloudEventFormContainer;
