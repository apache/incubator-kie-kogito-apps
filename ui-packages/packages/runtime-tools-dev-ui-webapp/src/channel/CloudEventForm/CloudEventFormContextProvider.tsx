import React, { useMemo } from 'react';
import {
  DevUIAppContext,
  useDevUIAppContext
} from '../../components/contexts/DevUIAppContext';
import CloudEventFormContext from './CloudEventFormContext';
import { CloudEventFormGatewayApiImpl } from './CloudEventFormGatewayApi';

interface IOwnProps {
  children;
}

const CloudEventFormContextProvider: React.FC<IOwnProps> = ({ children }) => {
  const runtimeToolsApi: DevUIAppContext = useDevUIAppContext();

  const gatewayApi = useMemo(
    () => new CloudEventFormGatewayApiImpl(runtimeToolsApi.getDevUIUrl()),
    []
  );

  return (
    <CloudEventFormContext.Provider value={gatewayApi}>
      {children}
    </CloudEventFormContext.Provider>
  );
};

export default CloudEventFormContextProvider;
