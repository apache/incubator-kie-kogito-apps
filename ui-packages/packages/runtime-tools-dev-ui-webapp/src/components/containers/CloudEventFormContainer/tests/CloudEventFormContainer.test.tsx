/*
 * Copyright 2023 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { mount } from 'enzyme';
import CloudEventFormContainer, {
  CloudEventFormContainerParams
} from '../CloudEventFormContainer';
import { CloudEventFormGatewayApiImpl } from '../../../../channel/CloudEventForm/CloudEventFormGatewayApi';
import * as CloudEventFormContext from '../../../../channel/CloudEventForm/CloudEventFormContext';
import { EmbeddedCloudEventForm } from '@kogito-apps/cloud-event-form/dist/embedded';

const routerParams: CloudEventFormContainerParams = {};

jest
  .spyOn(CloudEventFormContext, 'useCloudEventFormGatewayApi')
  .mockImplementation(
    () => new CloudEventFormGatewayApiImpl('http://localhost:9000')
  );

jest.mock('react-router', () =>
  Object.assign({}, jest.requireActual('@patternfly/react-icons'), {
    ...jest.requireActual('react-router'),
    useParams: () => routerParams
  })
);

const MockedComponent = (): React.ReactElement => {
  return <></>;
};

jest.mock('@kogito-apps/cloud-event-form/dist/embedded', () =>
  Object.assign(
    {},
    jest.requireActual('@kogito-apps/cloud-event-form/dist/embedded'),
    {
      EmbeddedCloudEventForm: () => {
        return <MockedComponent />;
      }
    }
  )
);

const properties = {
  isTriggerNewInstance: false,
  onSuccess: jest.fn(),
  onError: jest.fn()
};

describe('CloudEventFormContainer tests', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    delete routerParams.instanceId;
  });

  it('Snapshot', () => {
    const wrapper = mount(<CloudEventFormContainer {...properties} />).find(
      CloudEventFormContainer
    );

    expect(wrapper).toMatchSnapshot();

    const embeddedForm = wrapper.find(EmbeddedCloudEventForm);
    expect(embeddedForm.props().isNewInstanceEvent).toBeFalsy();
    expect(embeddedForm.props().driver).not.toBeUndefined();
    expect(
      embeddedForm.props().defaultValues.cloudEventSource
    ).not.toBeUndefined();
    expect(embeddedForm.props().defaultValues.instanceId).toBeUndefined();
  });

  it('Snapshot - with router param', () => {
    routerParams.instanceId = '1234';

    const wrapper = mount(<CloudEventFormContainer {...properties} />).find(
      CloudEventFormContainer
    );

    expect(wrapper).toMatchSnapshot();

    const embeddedForm = wrapper.find(EmbeddedCloudEventForm);
    expect(embeddedForm.props().isNewInstanceEvent).toBeFalsy();
    expect(embeddedForm.props().driver).not.toBeUndefined();
    expect(
      embeddedForm.props().defaultValues.cloudEventSource
    ).not.toBeUndefined();
    expect(embeddedForm.props().defaultValues.instanceId).toBe('1234');
  });
});
