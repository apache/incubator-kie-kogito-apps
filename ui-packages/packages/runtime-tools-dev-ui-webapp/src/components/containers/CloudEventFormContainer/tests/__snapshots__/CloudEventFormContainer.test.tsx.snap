// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloudEventFormContainer tests Snapshot - with router param 1`] = `
<CloudEventFormContainer
  isTriggerNewInstance={false}
  onError={[MockFunction]}
  onSuccess={[MockFunction]}
>
  <EmbeddedCloudEventForm
    data-ouia-component-type="cloud-event-form-container"
    data-ouia-safe={true}
    defaultValues={
      Object {
        "cloudEventSource": "/local/quarkus-devUi",
        "instanceId": "1234",
      }
    }
    driver={
      Object {
        "triggerCloudEvent": [Function],
      }
    }
    isNewInstanceEvent={false}
    targetOrigin="*"
  >
    <MockedComponent />
  </EmbeddedCloudEventForm>
</CloudEventFormContainer>
`;

exports[`CloudEventFormContainer tests Snapshot 1`] = `
<CloudEventFormContainer
  isTriggerNewInstance={false}
  onError={[MockFunction]}
  onSuccess={[MockFunction]}
>
  <EmbeddedCloudEventForm
    data-ouia-component-type="cloud-event-form-container"
    data-ouia-safe={true}
    defaultValues={
      Object {
        "cloudEventSource": "/local/quarkus-devUi",
        "instanceId": undefined,
      }
    }
    driver={
      Object {
        "triggerCloudEvent": [Function],
      }
    }
    isNewInstanceEvent={false}
    targetOrigin="*"
  >
    <MockedComponent />
  </EmbeddedCloudEventForm>
</CloudEventFormContainer>
`;
